File: CarRental/CarRental/model/Car.cs
using System.ComponentModel.DataAnnotations;

namespace application.model
{
  public class Car
  {
    [Key]
    public int CarId { get; set; } // Updated property name for consistency

    public int Model { get; set; }

    public string CarName { get; set; }

    public int Rentalprice { get; set; }

    public string CarImage { get; set; } // Added property to store the image filename

    public bool IsAvailable { get; set; } // New property to track availability
  }

  public class CarDTO
  {
    public string CarName { get; set; }
    public int Model { get; set; }
    public int Rentalprice { get; set; }
    public IFormFile ImageFile { get; set; } // This will be the image file uploaded by the user
  }

  public class CarUpdateDTO
  {
    public int CarId { get; set; } // ID of the car being updated
    public string CarName { get; set; }
    public int Model { get; set; }
    public int Rentalprice { get; set; }
    public IFormFile ImageFile { get; set; } // Optional image file for updating
    public string CarImage { get; set; } // To store the existing image path if not updating

    public bool? IsAvailable { get; set; } // Optional property to update availability
  }
}

File: UI/CarRental/src/app/car.ts
export interface Car {
  carId: number;
  model: number;
  carName: string;
  fileId: number;
  rentalprice: number;
  carImage: string;
  isAvailable: boolean;
}

export interface CarDTO {
  carName: string;
  model: number;
  rentalprice: number;
  imageFile: File; // File type for image
}

export interface CarUpdateDTO {
  carId: number;
  carName: string;
  model: number;
  rentalprice: number;
  imageFile?: File; // Optional file for update
  carImage?: string; // Existing image path, optional
  isAvailable?: boolean; // Optional availability
}
